lib/howl/interactions/external_command.moon

```diff
@@ -0,0 +1,207 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+glib = require 'ljglibs.glib'
+import app, interact, sys from howl
+import File from howl.io
+import Matcher from howl.util
+import ListWidget, markup from howl.ui
+import file_matcher, parse_path from howl.interactions.file_selection
+append = table.insert
+
+available_commands = ->
+  commands = {}
+  for path in sys.env.PATH\gmatch '[^:]+'
+    dir = File path
+    if dir.exists
+      append commands, child.basename for child in *dir.children
+
+  return commands
+
+text_parts = (text) ->
+  parts = [p for p in text\gmatch '%S+']
+  append parts, '' if text\ends_with ' '
+  parts
+
+last_text_part = (text) ->
+  parts = text_parts text
+  parts[#parts]
+
+get_cwd = ->
+  buffer = app.editor and app.editor.buffer
+  directory = buffer and (buffer.file and buffer.file.parent or buffer.directory)
+  directory or File glib.get_current_dir!
+
+looks_like_path = (text) ->
+  return unless text
+  for p in *{
```

Comment:
The dots here are not escaped, matching any character instead of the intended `.`.